{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Here I search for other known GFP variants, e.g. eGFP, sfGFP and also different colors. They will be tested for homology in blast to find mutations/differences in the sequences.\n",
    "# Afterwards I look at the original dataframe if we have these mutations and how the amino acid properties have changed.\n",
    "# It is possible to connect Bio.Blast with Biopython"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m                         Traceback (most recent call last)",
      "Cell \u001B[1;32mIn[7], line 4\u001B[0m\n\u001B[0;32m      1\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mBio\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mBlast\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m NCBIWWW, NCBIXML\n\u001B[0;32m      3\u001B[0m \u001B[38;5;66;03m# Perform a protein BLAST search\u001B[39;00m\n\u001B[1;32m----> 4\u001B[0m result_handle \u001B[38;5;241m=\u001B[39m \u001B[43mNCBIWWW\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mqblast\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mblastp\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mnr\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43msequence\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[0;32m      6\u001B[0m \u001B[38;5;66;03m# Parse the result handle\u001B[39;00m\n\u001B[0;32m      7\u001B[0m blast_records \u001B[38;5;241m=\u001B[39m NCBIXML\u001B[38;5;241m.\u001B[39mparse(result_handle)\n",
      "File \u001B[1;32m~\\Anaconda dl\\envs\\Data_science_RARL\\lib\\site-packages\\Bio\\Blast\\NCBIWWW.py:268\u001B[0m, in \u001B[0;36mqblast\u001B[1;34m(program, database, sequence, url_base, auto_format, composition_based_statistics, db_genetic_code, endpoints, entrez_query, expect, filter, gapcosts, genetic_code, hitlist_size, i_thresh, layout, lcase_mask, matrix_name, nucl_penalty, nucl_reward, other_advanced, perc_ident, phi_pattern, query_file, query_believe_defline, query_from, query_to, searchsp_eff, service, threshold, ungapped_alignment, word_size, short_query, alignments, alignment_view, descriptions, entrez_links_new_window, expect_low, expect_high, format_entrez_query, format_object, format_type, ncbi_gi, results_file, show_overview, megablast, template_type, template_length, username, password)\u001B[0m\n\u001B[0;32m    266\u001B[0m wait \u001B[38;5;241m=\u001B[39m qblast\u001B[38;5;241m.\u001B[39m_previous \u001B[38;5;241m+\u001B[39m delay \u001B[38;5;241m-\u001B[39m current\n\u001B[0;32m    267\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m wait \u001B[38;5;241m>\u001B[39m \u001B[38;5;241m0\u001B[39m:\n\u001B[1;32m--> 268\u001B[0m     \u001B[43mtime\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43msleep\u001B[49m\u001B[43m(\u001B[49m\u001B[43mwait\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    269\u001B[0m     qblast\u001B[38;5;241m.\u001B[39m_previous \u001B[38;5;241m=\u001B[39m current \u001B[38;5;241m+\u001B[39m wait\n\u001B[0;32m    270\u001B[0m \u001B[38;5;28;01melse\u001B[39;00m:\n",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m: "
     ]
    }
   ],
   "source": [
    "from Bio.Blast import NCBIWWW, NCBIXML\n",
    "\n",
    "# Performs a protein BLAST search\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", \"sequence\")\n",
    "\n",
    "# Parses the result handle\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "\n",
    "# Processes the results\n",
    "for blast_record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            print(\"****Alignment****\")\n",
    "            print(\"Sequence:\", alignment.title)\n",
    "            print(\"E-value:\", hsp.expect)\n",
    "            print(hsp.query[0:75] + \"...\")\n",
    "            print(hsp.match[0:75] + \"...\")\n",
    "            print(hsp.sbjct[0:75] + \"...\")\n",
    "# NCBIWWW.qblast() is used to perform a protein BLAST search\n",
    "\n",
    "# The parameters passed to the function are the BLAST program (blastp for protein), the database to search (nr for the non-redundant protein database), and the sequence you want to search.\n",
    "\n",
    "# The result_handle contains the XML results of the BLAST search.\n",
    "\n",
    "# NCBIXML.parse() function is used to parse the XML and obtain the blast_records. --> parse = analysieren/zergliedern\n",
    "\n",
    "# Iterate over the blast_records and extract the relevant information from the alignments and high-scoring pairs (HSPs).\n",
    "\n",
    "# Note that performing a BLAST search can take some time, depending on the size of the sequence and the complexity of the search. Also, keep in mind any usage restrictions and guidelines set by the NCBI when using their services.\n",
    "\n",
    "\n",
    "# Expected output:\n",
    "# ****Alignment****\n",
    "# Sequence: [Alignment Title]\n",
    "# E-value: [E-value]\n",
    "# [Query sequence snippet]...\n",
    "# [Match sequence snippet]...\n",
    "# [Subject sequence snippet]...\n"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:43:34.875688200Z",
     "start_time": "2023-07-09T14:42:32.082044600Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                             Sequence  \\\n",
      "0   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "1   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "2   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "3   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "4   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "5   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "6   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "7   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "8   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "9   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "10  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "11  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "12  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "13  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "14  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "15  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "16  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "17  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "18  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "19  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "20  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "21  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "22  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "23  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "24  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "25  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "26  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "27  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "28  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "29  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "30  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "31  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "32  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "33  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "34  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "35  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "36  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "37  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "38  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "39  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "40  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "41  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "42  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "43  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "44  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "45  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "46  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "47  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "48  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "49  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "\n",
      "                                      Alignment Title        E-value  \\\n",
      "0   gb|ABF18931.1| green fluorescent protein, part...  3.485920e-177   \n",
      "1   gb|AHE38523.1| linker-GFP, partial [Cloning ve...  2.683210e-176   \n",
      "2   gb|QQL06402.1| myrGFP from pTaDaG2 [Cloning ve...  3.340480e-176   \n",
      "3   gb|AAB51347.1| green fluorescent protein [synt...  3.400920e-176   \n",
      "4   dbj|BAJ22064.1| unnamed protein product [Binar...  4.348030e-176   \n",
      "5   ref|WP_217469231.1| hypothetical protein, part...  4.359380e-176   \n",
      "6   ref|WP_174827543.1| hypothetical protein [Stap...  4.471870e-176   \n",
      "7   sp|P42212.1| RecName: Full=Green fluorescent p...  8.176940e-176   \n",
      "8   gb|ADW65650.2| green fluorescent protein 2454-...  9.401680e-176   \n",
      "9   gb|AAG29470.1| GFP[AAV] [Promoter probe vector...  1.080440e-175   \n",
      "10  dbj|BBG75473.1| mitochondria-targeted syntheti...  1.203350e-175   \n",
      "11  gb|AAG29478.1| GFP[ASV] [Promoter probe vector...  1.219030e-175   \n",
      "12         gb|AZQ25072.1| yEGFP [synthetic construct]  1.404240e-175   \n",
      "13  gb|AAL17988.1| green fluorescent protein [Expr...  1.743440e-175   \n",
      "14  ref|WP_087671979.1| MULTISPECIES: hypothetical...  1.882630e-175   \n",
      "15  gb|AAG29486.1| GFP[LVA] [Promoter probe vector...  1.890650e-175   \n",
      "16  gb|ABL60857.1| GFP [Expression vector pCeMM-NT...  2.040490e-175   \n",
      "17  dbj|BBD84636.1| enhanced green fluorescent pro...  2.320860e-175   \n",
      "18  gb|AAD55053.1| green fluorescent protein [Expr...  2.370500e-175   \n",
      "19  gb|ACO90416.1| GFP-SsrA(SP) [synthetic construct]  2.975820e-175   \n",
      "20  gb|AEP13835.1| GFPmut1 [Cloning vector pVLG6] ...  3.084680e-175   \n",
      "21   gb|AHI43152.1| GFP [Gene trapping vector pDB790]  3.187910e-175   \n",
      "22  dbj|BAB68313.1| codon-optimized GFPmut1 [Expre...  3.295060e-175   \n",
      "23  gb|QPP46746.1| green fluorescent protein [synt...  3.455200e-175   \n",
      "24  gb|AEX63422.1| monomeric yeast-enhanced green ...  3.480410e-175   \n",
      "25  ref|WP_273818479.1| hypothetical protein [Esch...  3.732430e-175   \n",
      "26  gb|AAB47852.1| green fluorescent protein [synt...  4.103090e-175   \n",
      "27  gb|ABN41558.1| gfp/ova fusion [Promoter-trap v...  4.212250e-175   \n",
      "28  gb|AAA69550.1| green fluorescent protein [Clon...  4.466020e-175   \n",
      "29  gb|UHA47092.1| GFP [Transformation vector BAC ...  4.498190e-175   \n",
      "30  dbj|GLH76640.1| hypothetical protein SSBR45G_1...  4.564500e-175   \n",
      "31  dbj|BAF90739.1| green fluorescent protein, par...  5.280990e-175   \n",
      "32  gb|ADZ04469.1| sGFP(S65T)-ER [Binary vector pB...  5.322910e-175   \n",
      "33  dbj|BBG75461.1| er-targeted synthetic green fl...  5.380220e-175   \n",
      "34  gb|AAS76482.1| actin/green fluorescent protein...  6.320860e-175   \n",
      "35           gb|QAA95701.1| GFP [synthetic construct]  6.505160e-175   \n",
      "36  gb|AOC59748.1| sGFP65T [Cloning vector GmScrea...  6.747780e-175   \n",
      "37  dbj|BBG75465.1| nucleus-targeted synthetic gre...  7.957510e-175   \n",
      "38  gb|ASM41183.1| sGFP-GPI fusion protein [Expres...  7.958780e-175   \n",
      "39  gb|AAF04100.1| green fluorescent protein [synt...  8.190900e-175   \n",
      "40  dbj|BBG75469.1| peroxisome-targeted synthetic ...  9.075560e-175   \n",
      "41  gb|ADF80258.1| green fluorescent protein [Clon...  9.110810e-175   \n",
      "42  gb|ABP96788.1| green fluorescent protein [Clon...  9.245850e-175   \n",
      "43  gb|AAB51348.1| green fluorescent protein [synt...  9.659320e-175   \n",
      "44  gb|ACS28253.1| green fluorescent protein [Expr...  9.694980e-175   \n",
      "45  gb|AAG26344.1| Cox8ND6gfp fusion protein [synt...  9.971220e-175   \n",
      "46  gb|QEQ12685.1| enhanced green fluorescent prot...  1.062050e-174   \n",
      "47  gb|AFA52650.1| eGFP [synthetic construct] >gb|...  1.101530e-174   \n",
      "48  ref|WP_012569506.1| MULTISPECIES: green fluore...  1.138390e-174   \n",
      "49  dbj|BCY04410.1| green fluorescent protein [Bin...  1.138520e-174   \n",
      "\n",
      "                                       Query Sequence  \\\n",
      "0   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "1   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "2   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "3   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "4   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "5   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "6   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "7   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "8   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "9   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "10  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "11  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "12  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "13  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "14  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "15  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "16  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "17  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "18  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "19  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "20  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "21  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "22  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "23  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "24  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "25  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "26  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "27  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "28  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "29  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "30  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "31  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "32  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "33  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "34  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "35  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "36  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "37  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "38  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "39  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "40  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "41  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "42  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "43  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "44  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "45  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "46  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "47  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "48  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "49  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "\n",
      "                                       Match Sequence  \\\n",
      "0   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "1   +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "2   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "3   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "4   +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "5   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "6   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "7   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "8   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "9   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "10  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "11  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "12  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "13  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "14  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "15  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "16  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "17  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "18  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "19  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "20  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "21  MSKGEELFTGVVPILVELDGDVNG+KFSVSGEGEGDATYGKLTLKF...   \n",
      "22  M KGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "23  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "24  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "25  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "26  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "27  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "28  M KGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "29  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "30  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "31  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "32  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "33  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "34  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "35  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "36  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "37  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "38  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "39  M+KGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "40  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "41  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "42  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "43  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "44  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "45  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "46  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "47  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "48  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "49  +SKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...   \n",
      "\n",
      "                                     Subject Sequence  \n",
      "0   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "1   VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "2   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "3   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "4   VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "5   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "6   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "7   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "8   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "9   MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "10  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "11  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "12  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "13  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "14  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "15  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "16  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "17  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "18  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "19  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "20  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "21  MSKGEELFTGVVPILVELDGDVNGYKFSVSGEGEGDATYGKLTLKF...  \n",
      "22  MRKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "23  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "24  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "25  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "26  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "27  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "28  MGKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "29  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "30  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "31  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "32  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "33  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "34  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "35  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "36  LSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "37  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "38  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "39  MAKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "40  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "41  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "42  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "43  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "44  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "45  LSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "46  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "47  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "48  MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n",
      "49  VSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF...  \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n",
      "C:\\Users\\roman\\AppData\\Local\\Temp\\ipykernel_29180\\1394206210.py:30: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  df_results1 = df_results1.append(result, ignore_index=True)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio.Blast import NCBIWWW, NCBIXML\n",
    "\n",
    "unmutated_sequence = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "\n",
    "sequence_to_search = unmutated_sequence\n",
    "\n",
    "\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", sequence_to_search)\n",
    "\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "\n",
    "# Create an empty DataFrame to store the results\n",
    "df_results1 = pd.DataFrame(\n",
    "    columns=[\"Sequence\", \"Alignment Title\", \"E-value\", \"Query Sequence\", \"Match Sequence\", \"Subject Sequence\"])\n",
    "\n",
    "# Process the results and store in the DataFrame\n",
    "for blast_record in blast_records:\n",
    "    for alignment in blast_record.alignments:\n",
    "        for hsp in alignment.hsps:\n",
    "            result = {\n",
    "                \"Sequence\": sequence_to_search,\n",
    "                \"Alignment Title\": alignment.title,\n",
    "                \"E-value\": hsp.expect,\n",
    "                \"Query Sequence\": hsp.query,\n",
    "                \"Match Sequence\": hsp.match,\n",
    "                \"Subject Sequence\": hsp.sbjct\n",
    "            }\n",
    "            df_results1 = df_results1.append(result, ignore_index=True)\n",
    "\n",
    "print(df_results1)\n",
    "\n",
    "# The display of the results is very confusing and unorganized.\n",
    "# Additionally, configuring all the search parameters is very complicated. It might be easier to use the Blast-Website and then import the data."
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:46:38.778616500Z",
     "start_time": "2023-07-09T14:45:36.536664300Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "#FPbase is a really good website for all GFP-proteins\n",
    "\n",
    "Unmutiert:\n",
    ">unmutated_GFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "#We do not have a completely unmutated sequence, because we already have the F64L mutation in comparison to avGFP.\n",
    "\n",
    "eGFP: Tsien RY (1998) The green fluorescent protein. Annu Rev Biochem 67: 509â€“544.\n",
    ">eGFP_1\n",
    "MVSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITLGMDELYK\n",
    "\n",
    "sfGFP: https://www.fpbase.org/protein/superfolder-gfp/\n",
    ">sfGFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "\n",
    "alphaGFP: https://www.fpbase.org/protein/alphagfp/\n",
    ">alphaGFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "\n",
    "avGFP: https://www.fpbase.org/protein/avgfp/\n",
    ">avGFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gb|QEG99026.1| codon-optimized 3XeGFP [Cloning vector pTrichoGate-6]\n",
      "gb|AEG42740.1| SV40-3xeGFP [Cloning vector pPLV04] >gb|AIQ78254.1| n3xGFP [Cloning vector pPLV04_v2]\n",
      "gb|QNN02187.1| 3xVenus-SYP [Cloning vector PjCADL1pro::3xVenus-SYP] >gb|QNN02189.1| 3xVenus-SYP [Cloning vector PjCADL2pro::3xVenus-SYP] >gb|QNN02191.1| 3xVenus-SYP [Cloning vector PjCADL3pro::3xVenus-SYP] >gb|QNN02193.1| 3xVenus-SYP [Cloning vector PjCADL4pro::3xVenus-SYP] >gb|QNN02195.1| 3xVenus-SYP [Cloning vector PjCADL5pro::3xVenus-SYP]\n",
      "gb|AHE38517.1| 3x GFP, partial [Cloning vector pGGC025]\n",
      "dbj|BAS49686.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB123] >dbj|BAS49690.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB124] >dbj|BAS49694.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB125] >dbj|BAS49821.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB223] >dbj|BAS49825.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB224] >dbj|BAS49829.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB225] >dbj|BAS49956.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB323] >dbj|BAS49960.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB324] >dbj|BAS49964.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB325] >dbj|BAS50091.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB423] >dbj|BAS50095.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB424] >dbj|BAS50099.1| three repeats of Citrine with GGSGGS linkers [Binary vector pMpGWB425]\n",
      "gb|QBF54091.1| integrin beta 1/meGFP fusion protein [Cloning vector PBQP-IntB1-EGFP2x]\n",
      "gb|QBF54089.1| core histone H2B/meGFP fusion protein [Cloning vector PBQP-H2B-EGFP2x]\n",
      "gb|UNO37905.1| EYFP [Cloning vector pTubEYFP]\n",
      "gb|AGW21605.1| NLS-YFP-CFP [Yeast integrative vector pBS42]\n",
      "gb|AGW21608.1| NLS-YFPxCFP [Yeast integrative vector pBS42BN]\n",
      "gb|AGW21614.1| NLS-YFP-Pro5-CFP [Yeast integrative vector pBS46]\n",
      "gb|AGW21617.1| NLS-YFP-Pro10-CFP [Yeast integrative vector pBS47]\n",
      "gb|AGW21620.1| NLS-YFP-Pro15-CFP [Yeast integrative vector pBS48]\n",
      "gb|AEL12177.1| mVenus(L68V)-mTurquoise [synthetic construct]\n",
      "gb|AGW21626.1| NLS-YFP-Pro20-CFP [Yeast integrative vector pBS50]\n",
      "dbj|BAG31927.1| photoconvertible fluorescent protein Phamret [synthetic construct]\n",
      "gb|UXX23954.1| RAPGEF3-derived silenced cAMP biosensor cAMPFIRE-L-R279E, partial [synthetic construct]\n",
      "gb|UXX23951.1| RAPGEF3-derived cAMP biosensor cAMPFIRE-L, partial [synthetic construct]\n",
      "gb|UXX23952.1| RAPGEF3-derived cAMP biosensor cAMPFIRE-M, partial [synthetic construct]\n",
      "gb|UXX23953.1| RAPGEF3-derived cAMP biosensor cAMPFIRE-H, partial [synthetic construct]\n",
      "gb|ACB38271.1| MT1-MMP FRET probe protein, partial [synthetic construct]\n",
      "dbj|BAX25172.1| genetically encoded ratiometric fluorescent temperature indicator [synthetic construct]\n",
      "gb|ABF21065.1| calcium sensor cameleon D2cpv [synthetic construct]\n",
      "gb|UBZ25896.1| tdTomato-p2A-EGFP-Syb2 [Cloning vector pAAV-SynaptoTAG2]\n",
      "gb|WEQ72081.1| msfGFP-mRuby3-F3SG [synthetic construct]\n",
      "gb|WEQ72082.1| msfGFP-mScarletF3SG [synthetic construct]\n",
      "gb|WEQ72084.1| msfGFP-tagRFP-T-F3SG [synthetic construct]\n",
      "gb|WEQ72085.1| msfGFP-stag-F3SG [synthetic construct]\n",
      "gb|WEQ72078.1| msfGFP-FRMVQ-F3SG [synthetic construct]\n",
      "gb|WEQ72083.1| msfGFP-mStrawberry-F3SG [synthetic construct]\n",
      "gb|WEQ72080.1| msfGFP-mOrange2-F3SG [synthetic construct]\n",
      "dbj|BBA16890.1| Venus-AG [synthetic construct]\n",
      "gb|AQY63022.1| EBFP2-I-PhoC1-HCVp [Expression vector pcDNA-I-PhoCl-HCVp]\n",
      "dbj|BBM34117.1| signal retaining autophagy indicator [synthetic construct]\n",
      "gb|WEQ72079.1| msfGFP-mCherry-F3SG [synthetic construct]\n",
      "dbj|BBM34118.1| mito-signal retaining autophagy indicator [synthetic construct]\n",
      "dbj|BAS32675.1| mRuby2(delC12)-mClover(delC10)-RLuc8-S257G [synthetic construct]\n",
      "gb|AAX13969.1| GFP-DsRed fusion protein [Dual fluorescent protein cloning vector pGRFP]\n",
      "dbj|BAS32670.1| mKusabiraOrange2-Yellow-Nano-lantern [synthetic construct]\n",
      "dbj|BAS32671.1| mKusabiraOrange2-Yellow-Nano-lantern(delN3) [synthetic construct]\n",
      "ref|WP_270022569.1| hypothetical protein [Pseudomonas aeruginosa] >gb|MCZ9677143.1| hypothetical protein [Pseudomonas aeruginosa]\n",
      "dbj|BAS32669.1| mKusabiraOrange2-2xGGGGS-Yellow-Nano-lantern [synthetic construct]\n",
      "dbj|BAS32676.1| mRuby2(delC12)-mClover(delN11delC10)-RLuc8-S257G [synthetic construct]\n",
      "gb|ATA67138.1| YPet(short)-stHP35-mCherry tension sensor [synthetic construct]\n",
      "gb|ATA67137.1| YPet(short)-HP35-mCherry tension sensor [synthetic construct]\n",
      "gb|AAF62891.1| enhanced GFP variant [Cloning vector pEGAD]\n",
      "gb|AUR53415.1| GFPBteAAV [Cloning vector pHT-gfpBteAAV]\n",
      "gb|QMV47820.1| EGFP [Cloning vector AGG1521:pCHIKVRep1]\n",
      "gb|AAB02576.1| enhanced green fluorescent protein [Cloning vector pEGFP-C1] >gb|AAG27429.1| enhanced green fluorescent protein [Cloning vector pdeltaE1sp1A(CMV-GFP)] >gb|ABQ43174.1| EGFP fusion protein [Plant transit expression vector pSAT6-EGFP-Ca] >dbj|BDT06276.1| green fluorescent protein [Expression vector YHp22601]\n",
      "emb|SCV58932.1| Green fluorescent protein [Fusarium fujikuroi]\n"
     ]
    }
   ],
   "source": [
    "#BLAST IN PYTHON\n",
    "from Bio.Blast import NCBIWWW\n",
    "from Bio.Blast import NCBIXML\n",
    "\n",
    "unmutated_GFP_1 = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "eGFP = \"MVSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITLGMDELYK\"\n",
    "sfGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "alphaGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "avGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\",\n",
    "                               unmutated_GFP_1 + \"\\n\" + eGFP + \"\\n\" + sfGFP + \"\\n\" + alphaGFP + \"\\n\" + avGFP)\n",
    "blast_records = NCBIXML.parse(result_handle)\n",
    "for record in blast_records:\n",
    "    for alignment in record.alignments:\n",
    "        # Extract information from the alignment\n",
    "        print(alignment.title)\n",
    "# Searches the entire database, which is not exactly what I want."
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:48:48.543363200Z",
     "start_time": "2023-07-09T14:47:45.898413300Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Next try: Use clustal omega or muscle because they are better suited for an offline multiple alignment and they achieve a higher accuracy.\n"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "2. Try: in Clustal omega website, because integrating python is unnecessarily complicated...\n",
    "Job ID: clustalo-I20230619-103006-0921-81728093-p1m\n",
    "https://www.ebi.ac.uk/Tools/services/web/toolresult.ebi?jobId=clustalo-I20230619-103006-0921-81728093-p1m\n",
    "CLUSTAL O(1.2.4) multiple sequence alignment\n",
    "\n",
    "\n",
    "sfGFP_1              -MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPT\t59\n",
    "alphaGFP_1           -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\t59\n",
    "eGFP_1               MVSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\t60\n",
    "unmutated_GFP_1      -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\t59\n",
    "avGFP_1              -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\t59\n",
    "                      :**************************** ******** ********************\n",
    "\n",
    "sfGFP_1              LVTTLTYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTL\t119\n",
    "alphaGFP_1           LVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTL\t119\n",
    "eGFP_1               LVTTLTYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\t120\n",
    "unmutated_GFP_1      LVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\t119\n",
    "avGFP_1              LVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\t119\n",
    "                     ****::**************:****************** *****.**************\n",
    "\n",
    "sfGFP_1              VNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGSVQLA\t179\n",
    "alphaGFP_1           VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLA\t179\n",
    "eGFP_1               VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\t180\n",
    "unmutated_GFP_1      VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\t179\n",
    "avGFP_1              VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\t179\n",
    "                     *************************:******* *********.*******:********\n",
    "\n",
    "sfGFP_1              DHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\t238\n",
    "alphaGFP_1           DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\t238\n",
    "eGFP_1               DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITLGMDELYK\t239\n",
    "unmutated_GFP_1      DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\t238\n",
    "avGFP_1              DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\t238\n",
    "                     **************************.************************ *******\n",
    "Tree data\n",
    "(\n",
    "(\n",
    "(\n",
    "sfGFP_1:0.02941,\n",
    "alphaGFP_1:0.00420)\n",
    ":0.01366,\n",
    "eGFP_1:0.01155)\n",
    ":0.00105,\n",
    "unmutated_GFP_1:0.00105,\n",
    "avGFP_1:0.00315);\n",
    "\n",
    "3. Try: in MUSCLE (is supposed to be faster and more accurate)\n",
    "muscle-I20230619-103332-0882-21245675-p1m\n",
    "https://www.ebi.ac.uk/Tools/services/web/toolresult.ebi?jobId=muscle-I20230619-103332-0882-21245675-p1m\n",
    "CLUSTAL multiple sequence alignment by MUSCLE (3.8)\n",
    "\n",
    "\n",
    "sfGFP_1              -MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPT\n",
    "alphaGFP_1           -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\n",
    "eGFP_1               MVSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\n",
    "unmutated_GFP_1      -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\n",
    "avGFP_1              -MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPT\n",
    "                      :**************************** ******** ********************\n",
    "\n",
    "sfGFP_1              LVTTLTYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTL\n",
    "alphaGFP_1           LVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTL\n",
    "eGFP_1               LVTTLTYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\n",
    "unmutated_GFP_1      LVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\n",
    "avGFP_1              LVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTL\n",
    "                     ****::**************.****************** *****.**************\n",
    "\n",
    "sfGFP_1              VNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGSVQLA\n",
    "alphaGFP_1           VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLA\n",
    "eGFP_1               VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\n",
    "unmutated_GFP_1      VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\n",
    "avGFP_1              VNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLA\n",
    "                     *************************:******* *********.*******:********\n",
    "\n",
    "sfGFP_1              DHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "alphaGFP_1           DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "eGFP_1               DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITLGMDELYK\n",
    "unmutated_GFP_1      DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "avGFP_1              DHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "                     **************************.************************ *******\n",
    " Tree data\n",
    "(\n",
    "(\n",
    "(\n",
    "sfGFP_1:0.02941,\n",
    "alphaGFP_1:0.00420)\n",
    ":0.01366,\n",
    "eGFP_1:0.01155)\n",
    ":0.00105,\n",
    "unmutated_GFP_1:0.00105,\n",
    "avGFP_1:0.00315);"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Next steps:\n",
    "#Calculate all the properties of all the GFP variants and compare the differences.\n",
    "#Check whether we have the mutations in our data set that lead from one variant to the other."
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Sequence Name  Molecular Weight  Residue Weight    pKa1     pKb2    pKx3  \\\n",
      "0  unmutated GFP          31122.34        26834.81  498.47  2133.09  746.86   \n",
      "1           eGFP          31229.54        26923.99  501.72  2142.86  736.79   \n",
      "2          sfGFP          31051.24        26763.72  502.89  2150.77  746.86   \n",
      "3          avGFP          31156.35        26868.83  497.94  2132.62  757.39   \n",
      "4       alphaGFP          31066.16        26778.65  497.64  2132.74  755.04   \n",
      "\n",
      "       pl4      H      VSC      P1      P2     SASA     NCISC  \n",
      "0  1412.64  -9.68  15894.1  1997.6  38.159  406.603  9.501487  \n",
      "1  1416.91  -7.28  16076.1  1996.9  38.570  409.399  9.600551  \n",
      "2  1422.51 -15.07  16105.9  2019.2  38.825  411.887  9.558542  \n",
      "3  1412.14  -9.55  15916.1  1997.9  38.263  406.900  9.487367  \n",
      "4  1417.35 -13.48  15693.4  2007.5  37.670  404.607  9.418697  \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Defines the protein sequences and their respective sequence names\n",
    "sequences = [unmutated_GFP_1, eGFP, sfGFP, avGFP, alphaGFP]\n",
    "sequence_names = [\"unmutated GFP\", \"eGFP\", \"sfGFP\", \"avGFP\", \"alphaGFP\"]\n",
    "\n",
    "ASE_datei = pd.read_csv(r\"C:\\Users\\roman\\Desktop\\AS Eigenschaften\\aminoacids.csv\")\n",
    "\n",
    "columns_to_delete = ['Name', 'Abbr', 'Molecular Formula', 'Residue Formula']\n",
    "ASE_datei.drop(columns=columns_to_delete, inplace=True)\n",
    "delete_columns = [\"carbon\", \"hydrogen\", \"nitrogen\", \"oxygen\", \"sulfur\"]\n",
    "ASE_datei.drop(columns=delete_columns, inplace=True)\n",
    "\n",
    "row_index_to_delete = 12\n",
    "ASE_clear = ASE_datei.drop(row_index_to_delete)\n",
    "\n",
    "# Creates a dictionary for each property mapping\n",
    "property_maps = {}\n",
    "for col in ASE_clear.columns:\n",
    "    if col != \"Letter\":  # Skip the \"Letter\" column\n",
    "        property_maps[col] = dict(zip(ASE_clear[\"Letter\"].dropna(), ASE_clear[col].dropna()))\n",
    "\n",
    "# Creates an empty DataFrame to store the results\n",
    "results_df = pd.DataFrame(columns=[\"Sequence Name\"] + list(property_maps.keys()))\n",
    "\n",
    "# Calculates the sum of products for each property for each sequence\n",
    "sequence_results = []\n",
    "for sequence, sequence_name in zip(sequences, sequence_names):\n",
    "    results = {}\n",
    "    for prop, property_map in property_maps.items():\n",
    "        result = 0\n",
    "        for AA in property_map.keys():\n",
    "            quantity = sequence.count(AA)\n",
    "            result += property_map[AA] * quantity\n",
    "        results[prop] = result\n",
    "\n",
    "\n",
    "    sequence_results.append(pd.DataFrame({\"Sequence Name\": sequence_name, **results}, index=[0]))\n",
    "\n",
    "# Concatenates all sequence results into the main results DataFrame\n",
    "results_MuSe_ASE_df = pd.concat(sequence_results, ignore_index=True)\n",
    "\n",
    "# Print the results DataFrame\n",
    "print(results_MuSe_ASE_df)\n"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:51:13.638853500Z",
     "start_time": "2023-07-09T14:51:13.590341200Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Colour variants of GFP:\n",
    "CFP: https://www.fpbase.org/protein/cfp/ Y66W\n",
    ">CFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSWGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "\n",
    "RFP: https://www.fpbase.org/protein/eqfp611/ native sequence from Entacmaea quadricolor\n",
    "> eqFP611_1\n",
    "MNSLIKENMRMMVVMEGSVNGYQFKCTGEGDGNPYMGTQTMRIKVVEGGPLPFAFDILATSFMYGSKTFIKHTKGIPDFFKQSFPEGFTWERVTRYEDGGVFTVMQDTSLEDGCLVYHAKVTGVNFPSNGAVMQKKTKGWEPNTEMLYPADGGLRGYSQMALNVDGGGYLSCSFETTYRSKKTVENFKMPGFHFVDHRLERLEESDKEMFVVQHEHAVAKFCDLPSKLGRL\n",
    "\n",
    "hyperfold_YFP: has a lot of mutations away from avGFP https://www.fpbase.org/protein/hyperfolder-yfp/\n",
    "> hfYFP_1\n",
    "MVSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKLISTTGKLPVPWPTLVTTLGYGLMVFARYPDHMKQHDFFKSAMPEGYVQERTISFEDDGYYKTRAEVKFEGDTLVNRIVLKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGGVQLADHYQQNTPIGDGPVLLPDNHYLSYQSVLSKDPNEKRDHMVLKERVTAAGITHDMNELYK\n",
    "\n",
    "avGFP: https://www.fpbase.org/protein/avgfp/\n",
    ">avGFP_1\n",
    "MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\n",
    "\n",
    "mOFP: https://www.fpbase.org/protein/morange/\n",
    ">mOFP_1\n",
    "MVSKGEENNMAIIKEFMRFKVRMEGSVNGHEFEIEGEGEGRPYEGFQTAKLKVTKGGPLPFAWDILSPQFTYGSKAYVKHPADIPDYFKLSFPEGFKWERVMNFEDGGVVTVTQDSSLQDGEFIYKVKLRGTNFPSDGPVMQKKTMGWEASSERMYPEDGALKGEIKMRLKLKDGGHYTSEVKTTYKAKKPVQLPGAYIVGIKLDITSHNEDYTIVEQYERAEGRHSTGGMDELYK"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "source": [
    "Results of the Muscle of the color variants:\n",
    "CFP_1           -MSKGEELFTGVVP----ILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF-ICTTGKLP\n",
    "avGFP_1         -MSKGEELFTGVVP----ILVELDGDVNGHKFSVSGEGEGDATYGKLTLKF-ICTTGKLP\n",
    "hfYFP_1         MVSKGEELFTGVVP----ILVELDGDVNGHKFSVRGEGEGDATNGKLTLKL-ISTTGKLP\n",
    "eqFP611_1       --------MNSLIKENMRMMVVMEGSVNGYQFKCTGEGDGNPYMGTQTMRIKVVEGGPLP\n",
    "mOFP_1          MVSKGEENNMAIIKEFMRFKVRMEGSVNGHEFEIEGEGEGRPYEGFQTAKLKVTKGGPLP\n",
    "                          .::     : * ::*.***::*.  ***:* .  *  * .: :   * **\n",
    "\n",
    "CFP_1           VPWPTLVTTFSWGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKF\n",
    "avGFP_1         VPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKF\n",
    "hfYFP_1         VPWPTLVTTLGYGLMVFARYPDHMKQHDFFKSAMPEGYVQERTISFEDDGYYKTRAEVKF\n",
    "eqFP611_1       FAFDILATSFMYGSKTFIKHTKGIP--DFFKQSFPEGFTWERVTRYEDGGVFTVMQDTSL\n",
    "mOFP_1          FAWDILSPQFTYGSKAYVKHPADIP--DYFKLSFPEGFKWERVMNFEDGGVVTVTQDSSL\n",
    "                ..:  * . : :*   : .:.  :   *:** ::***: .**.  ::*.*  ..  : .:\n",
    "\n",
    "CFP_1           EGDTLVNRIELKGIDFKEDGNILGHK-LEYNYNSHNVYIMADKQKNGIKVNFKIRHNIED\n",
    "avGFP_1         EGDTLVNRIELKGIDFKEDGNILGHK-LEYNYNSHNVYIMADKQKNGIKVNFKIRHNIED\n",
    "hfYFP_1         EGDTLVNRIVLKGIDFKEDGNILGHK-LEYNFNSHNVYITADKQKNGIKANFKIRHNVED\n",
    "eqFP611_1       EDGCLVYHAKVTGVNFPSNGAVMQKKTKGWEPNTEMLY----PADGGLRGYSQMALNVDG\n",
    "mOFP_1          QDGEFIYKVKLRGTNFPSDGPVMQKKTMGWEASSERMY----PEDGALKGEIKMRLKLKD\n",
    "                :.. :: .  : * :* .:* :: :*   :: .:  :*      ...:.   ::  ::..\n",
    "\n",
    "CFP_1           GS---VQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAA--G\n",
    "avGFP_1         GS---VQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAA--G\n",
    "hfYFP_1         GG---VQLADHYQQNTPIGDGPVLLPDNHYLSYQSVLSKDPNEKRDHMVLKERVTAA--G\n",
    "eqFP611_1       GGYLSCSFETTYRSKKTVEN--FKMPGFHFVDHRLERLEE-SDKEMFVVQHEHAVAKFCD\n",
    "mOFP_1          GGHYTSEVKTTYKAKKPVQ-----LPGAYIVGIKLDITSH-NEDYTIVEQYERAEGR--H\n",
    "                *.    ..   *. :..:      :*. : :. .    .  .:.   :   * . .\n",
    "\n",
    "CFP_1           ITHGMDELYK\n",
    "avGFP_1         ITHGMDELYK\n",
    "hfYFP_1         ITHDMNELYK\n",
    "eqFP611_1       LPSKLGRL--\n",
    "mOFP_1          STGGMDELYK\n",
    "                 .  :. *\n",
    "Tree data\n",
    "(\n",
    "CFP_1:0.00427,\n",
    "avGFP_1:0.00000,\n",
    "(\n",
    "hfYFP_1:0.04931,\n",
    "(\n",
    "eqFP611_1:0.28275,\n",
    "mOFP_1:0.21725)\n",
    ":0.42158)\n",
    ":0.05994);"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "outputs": [],
   "source": [
    "CFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSWGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "RFP = \"MNSLIKENMRMMVVMEGSVNGYQFKCTGEGDGNPYMGTQTMRIKVVEGGPLPFAFDILATSFMYGSKTFIKHTKGIPDFFKQSFPEGFTWERVTRYEDGGVFTVMQDTSLEDGCLVYHAKVTGVNFPSNGAVMQKKTKGWEPNTEMLYPADGGLRGYSQMALNVDGGGYLSCSFETTYRSKKTVENFKMPGFHFVDHRLERLEESDKEMFVVQHEHAVAKFCDLPSKLGRL\"\n",
    "hfYFP = \"MVSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKLISTTGKLPVPWPTLVTTLGYGLMVFARYPDHMKQHDFFKSAMPEGYVQERTISFEDDGYYKTRAEVKFEGDTLVNRIVLKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGGVQLADHYQQNTPIGDGPVLLPDNHYLSYQSVLSKDPNEKRDHMVLKERVTAAGITHDMNELYK\"\n",
    "avGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "mOFP = \"MVSKGEENNMAIIKEFMRFKVRMEGSVNGHEFEIEGEGEGRPYEGFQTAKLKVTKGGPLPFAWDILSPQFTYGSKAYVKHPADIPDYFKLSFPEGFKWERVMNFEDGGVVTVTQDSSLQDGEFIYKVKLRGTNFPSDGPVMQKKTMGWEASSERMYPEDGALKGEIKMRLKLKDGGHYTSEVKTTYKAKKPVQLPGAYIVGIKLDITSHNEDYTIVEQYERAEGRHSTGGMDELYK\"\n"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:52:47.217517800Z",
     "start_time": "2023-07-09T14:52:47.185374900Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  Sequence Name  Molecular Weight  Residue Weight    pKa1     pKb2    pKx3  \\\n",
      "0           CFP          31179.39        26891.87  500.14  2141.73  757.39   \n",
      "1           RFP          30196.58        26035.33  489.13  2084.61  780.22   \n",
      "2         hfYFP          31189.45        26883.87  503.83  2141.18  739.16   \n",
      "3         avGFP          31156.35        26868.83  497.94  2132.62  757.39   \n",
      "4          mOFP          30971.22        26719.88  495.49  2107.59  777.99   \n",
      "\n",
      "       pl4      H      VSC      P1      P2     SASA      NCISC  \n",
      "0  1412.37  -9.29  16033.4  2004.1  38.561  409.268   9.510966  \n",
      "1  1382.72  -3.91  15636.3  1900.4  38.150  399.483  10.010540  \n",
      "2  1431.27 -14.06  16167.3  2012.9  39.008  411.644   9.863915  \n",
      "3  1412.14  -9.55  15916.1  1997.9  38.263  406.900   9.487367  \n",
      "4  1424.67 -21.32  15462.6  1989.4  37.212  400.781  10.699294  \n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Defines the protein sequences and their respective sequence names\n",
    "sequences = [CFP, RFP, hfYFP, avGFP, mOFP]\n",
    "sequence_names = [\"CFP\", \"RFP\", \"hfYFP\", \"avGFP\", \"mOFP\"]\n",
    "\n",
    "ASE_datei = pd.read_csv(r\"C:\\Users\\roman\\Desktop\\AS Eigenschaften\\aminoacids.csv\")\n",
    "\n",
    "columns_to_delete = ['Name', 'Abbr', 'Molecular Formula', 'Residue Formula']\n",
    "ASE_datei.drop(columns=columns_to_delete, inplace=True)\n",
    "delete_columns = [\"carbon\", \"hydrogen\", \"nitrogen\", \"oxygen\", \"sulfur\"]\n",
    "ASE_datei.drop(columns=delete_columns, inplace=True)\n",
    "\n",
    "row_index_to_delete = 12\n",
    "ASE_clear = ASE_datei.drop(row_index_to_delete)\n",
    "\n",
    "# Creates a dictionary for each property mapping\n",
    "property_maps = {}\n",
    "for col in ASE_clear.columns:\n",
    "    if col != \"Letter\":  # Skip the \"Letter\" column\n",
    "        property_maps[col] = dict(zip(ASE_clear[\"Letter\"].dropna(), ASE_clear[col].dropna()))\n",
    "\n",
    "# Creates an empty DataFrame to store the results\n",
    "results_df = pd.DataFrame(columns=[\"Sequence Name\"] + list(property_maps.keys()))\n",
    "\n",
    "# Calculates the sum of products for each property for each sequence\n",
    "sequence_results = []\n",
    "for sequence, sequence_name in zip(sequences, sequence_names):\n",
    "    results = {}\n",
    "    for prop, property_map in property_maps.items():\n",
    "        result = 0\n",
    "        for AA in property_map.keys():\n",
    "            quantity = sequence.count(AA)\n",
    "            result += property_map[AA] * quantity\n",
    "        results[prop] = result\n",
    "\n",
    "    # Creates a new DataFrame row for the current sequence\n",
    "    sequence_results.append(pd.DataFrame({\"Sequence Name\": sequence_name, **results}, index=[0]))\n",
    "\n",
    "# Concatenates all sequence results into the main results DataFrame\n",
    "results_MuSe_ASE_Farben_df = pd.concat(sequence_results, ignore_index=True)\n",
    "\n",
    "print(results_MuSe_ASE_Farben_df)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:52:51.685024800Z",
     "start_time": "2023-07-09T14:52:51.639874800Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'aligned_sequences.fasta'",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mFileNotFoundError\u001B[0m                         Traceback (most recent call last)",
      "Cell \u001B[1;32mIn[16], line 34\u001B[0m\n\u001B[0;32m     31\u001B[0m clustalomega_cline\n\u001B[0;32m     33\u001B[0m \u001B[38;5;66;03m# Parse the aligned records\u001B[39;00m\n\u001B[1;32m---> 34\u001B[0m alignment \u001B[38;5;241m=\u001B[39m \u001B[43mAlignIO\u001B[49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mread\u001B[49m\u001B[43m(\u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43maligned_sequences.fasta\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m,\u001B[49m\u001B[43m \u001B[49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[38;5;124;43mfasta\u001B[39;49m\u001B[38;5;124;43m\"\u001B[39;49m\u001B[43m)\u001B[49m\n\u001B[0;32m     36\u001B[0m \u001B[38;5;66;03m# Convert the alignment to a DataFrame\u001B[39;00m\n\u001B[0;32m     37\u001B[0m df_clusral_results \u001B[38;5;241m=\u001B[39m pd\u001B[38;5;241m.\u001B[39mDataFrame(alignment)\n",
      "File \u001B[1;32m~\\Anaconda dl\\envs\\Data_science_RARL\\lib\\site-packages\\Bio\\AlignIO\\__init__.py:383\u001B[0m, in \u001B[0;36mread\u001B[1;34m(handle, format, seq_count)\u001B[0m\n\u001B[0;32m    381\u001B[0m iterator \u001B[38;5;241m=\u001B[39m parse(handle, \u001B[38;5;28mformat\u001B[39m, seq_count)\n\u001B[0;32m    382\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m--> 383\u001B[0m     alignment \u001B[38;5;241m=\u001B[39m \u001B[38;5;28;43mnext\u001B[39;49m\u001B[43m(\u001B[49m\u001B[43miterator\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    384\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mStopIteration\u001B[39;00m:\n\u001B[0;32m    385\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mValueError\u001B[39;00m(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mNo records found in handle\u001B[39m\u001B[38;5;124m\"\u001B[39m) \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;28mNone\u001B[39m\n",
      "File \u001B[1;32m~\\Anaconda dl\\envs\\Data_science_RARL\\lib\\site-packages\\Bio\\AlignIO\\__init__.py:322\u001B[0m, in \u001B[0;36mparse\u001B[1;34m(handle, format, seq_count)\u001B[0m\n\u001B[0;32m    319\u001B[0m \u001B[38;5;28;01mif\u001B[39;00m seq_count \u001B[38;5;129;01mis\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28;01mNone\u001B[39;00m \u001B[38;5;129;01mand\u001B[39;00m \u001B[38;5;129;01mnot\u001B[39;00m \u001B[38;5;28misinstance\u001B[39m(seq_count, \u001B[38;5;28mint\u001B[39m):\n\u001B[0;32m    320\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mTypeError\u001B[39;00m(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mNeed integer for seq_count (sequences per alignment)\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[1;32m--> 322\u001B[0m \u001B[38;5;28;01mwith\u001B[39;00m as_handle(handle) \u001B[38;5;28;01mas\u001B[39;00m fp:\n\u001B[0;32m    323\u001B[0m     \u001B[38;5;66;03m# Map the file format to a sequence iterator:\u001B[39;00m\n\u001B[0;32m    324\u001B[0m     \u001B[38;5;28;01mif\u001B[39;00m \u001B[38;5;28mformat\u001B[39m \u001B[38;5;129;01min\u001B[39;00m _FormatToIterator:\n\u001B[0;32m    325\u001B[0m         iterator_generator \u001B[38;5;241m=\u001B[39m _FormatToIterator[\u001B[38;5;28mformat\u001B[39m]\n",
      "File \u001B[1;32m~\\Anaconda dl\\envs\\Data_science_RARL\\lib\\contextlib.py:135\u001B[0m, in \u001B[0;36m_GeneratorContextManager.__enter__\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    133\u001B[0m \u001B[38;5;28;01mdel\u001B[39;00m \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39margs, \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mkwds, \u001B[38;5;28mself\u001B[39m\u001B[38;5;241m.\u001B[39mfunc\n\u001B[0;32m    134\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m--> 135\u001B[0m     \u001B[38;5;28;01mreturn\u001B[39;00m \u001B[38;5;28;43mnext\u001B[39;49m\u001B[43m(\u001B[49m\u001B[38;5;28;43mself\u001B[39;49m\u001B[38;5;241;43m.\u001B[39;49m\u001B[43mgen\u001B[49m\u001B[43m)\u001B[49m\n\u001B[0;32m    136\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mStopIteration\u001B[39;00m:\n\u001B[0;32m    137\u001B[0m     \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mRuntimeError\u001B[39;00m(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mgenerator didn\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mt yield\u001B[39m\u001B[38;5;124m\"\u001B[39m) \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;28mNone\u001B[39m\n",
      "File \u001B[1;32m~\\Anaconda dl\\envs\\Data_science_RARL\\lib\\site-packages\\Bio\\File.py:72\u001B[0m, in \u001B[0;36mas_handle\u001B[1;34m(handleish, mode, **kwargs)\u001B[0m\n\u001B[0;32m     30\u001B[0m \u001B[38;5;124mr\u001B[39m\u001B[38;5;124;03m\"\"\"Context manager to ensure we are using a handle.\u001B[39;00m\n\u001B[0;32m     31\u001B[0m \n\u001B[0;32m     32\u001B[0m \u001B[38;5;124;03mContext manager for arguments that can be passed to SeqIO and AlignIO read, write,\u001B[39;00m\n\u001B[1;32m   (...)\u001B[0m\n\u001B[0;32m     69\u001B[0m \n\u001B[0;32m     70\u001B[0m \u001B[38;5;124;03m\"\"\"\u001B[39;00m\n\u001B[0;32m     71\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m---> 72\u001B[0m     \u001B[38;5;28;01mwith\u001B[39;00m \u001B[38;5;28mopen\u001B[39m(handleish, mode, \u001B[38;5;241m*\u001B[39m\u001B[38;5;241m*\u001B[39mkwargs) \u001B[38;5;28;01mas\u001B[39;00m fp:\n\u001B[0;32m     73\u001B[0m         \u001B[38;5;28;01myield\u001B[39;00m fp\n\u001B[0;32m     74\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mTypeError\u001B[39;00m:\n",
      "\u001B[1;31mFileNotFoundError\u001B[0m: [Errno 2] No such file or directory: 'aligned_sequences.fasta'"
     ]
    }
   ],
   "source": [
    "from Bio import AlignIO, SeqIO\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Align.Applications import ClustalOmegaCommandline\n",
    "import pandas as pd\n",
    "\n",
    "unmutated_GFP_1 = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "eGFP = \"MVSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITLGMDELYK\"\n",
    "sfGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVRGEGEGDATNGKLTLKFICTTGKLPVPWPTLVTTLTYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGTYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNFNSHNVYITADKQKNGIKANFKIRHNVEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSVLSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "alphaGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKRHDFFKSAMPEGYVQERTISFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYITADKQKNGIKANFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK\"\n",
    "avGFP = \"MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNY\"\n",
    "\n",
    "# Create SeqRecord objects for each sequence\n",
    "sequences = [\n",
    "    SeqRecord(Seq(unmutated_GFP_1), id=\"unmutated GFP\"),\n",
    "    SeqRecord(Seq(eGFP), id=\"eGFP\"),\n",
    "    SeqRecord(Seq(sfGFP), id=\"sfGFP\"),\n",
    "    SeqRecord(Seq(alphaGFP), id=\"alphaGFP\"),\n",
    "    SeqRecord(Seq(avGFP), id=\"avGFP\")\n",
    "]\n",
    "\n",
    "# Write the sequences to a temporary file in FASTA format\n",
    "temp_fasta_file = \"sequences.fasta\"\n",
    "SeqIO.write(sequences, temp_fasta_file, \"fasta\")\n",
    "\n",
    "# Performs MSA using the Clustal Omega command-line tool\n",
    "clustalomega_cline = ClustalOmegaCommandline(infile=temp_fasta_file, outfile=\"aligned_sequences.fasta\", verbose=True,\n",
    "                                             auto=True)\n",
    "clustalomega_cline\n",
    "\n",
    "# Parses the aligned records\n",
    "alignment = AlignIO.read(\"aligned_sequences.fasta\", \"fasta\")\n",
    "\n",
    "# Converts the alignment to a DataFrame\n",
    "df_clusral_results = pd.DataFrame(alignment)\n",
    "print(df_clustal_results)\n",
    "\n",
    "# I have tried to work using a temporary fasta, but the code does not work because my computer does not identify the fasta.\n",
    "# I was not able to fix the code, but shifted the focus, as the website mentioned above contains all the GFP variant information.\n",
    "# I just wanted to learn how to run blast/clustal analysis in python."
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-07-09T14:55:50.564756700Z",
     "start_time": "2023-07-09T14:55:49.699583200Z"
    }
   }
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
